build.sbt

```diff
@@ -1,19 +1,22 @@
 import Dependencies._
 
-import sbtcrossproject.crossProject
+import sbtcrossproject.CrossPlugin.autoImport.crossProject
 import org.scalajs.jsenv.nodejs.NodeJSEnv
 
+Global / onChangedBuildSource := ReloadOnSourceChanges
+
 // Binary compatibility is this version
 val previousVersion: Option[String] = Some("1.5.0")
 
 val ScalaTestVersion = "3.2.10"
 
-// Do NOT upgrade these dependencies to 2.x or newer! twirl is a sbt-plugin
-// and gets published with Scala 2.12, therefore we need to stay at the same major version
-// like the 2.12.x Scala compiler, otherwise we run into conflicts when using sbt 1.5+
-// See https://github.com/scala/scala/pull/9743
-val ScalaParserCombinatorsVersion = "1.1.2" // Do not upgrade beyond 1.x
-val ScalaXmlVersion               = "1.3.0" // Do not upgrade beyond 1.x
+def parserCombinators(version: String) = {
+  CrossVersion.partialVersion(version) match {
+    case Some((2, _)) =>
+      Seq("org.scala-lang.modules" %% "scala-parser-combinators" % "1.1.2" % Optional)
+    case _ => Seq("org.scala-lang.modules" %% "scala-parser-combinators" % "2.1.0" % Optional)
+  }
+}
```

Comment:
I just commited a similiar solution to the `main` branch (https://github.com/playframework/twirl/commit/ab2821f82728b76accdc7a723326f2ada03d16bd), please rebase.

Reply:
I'll do that this morning!

Reply:
Done!